src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -298,6 +306,10 @@ public function updateAction(Request $request)
                 return $this->redirectHandler->redirectToResource($configuration, $resource);
             }
 
+            if ($configuration->hasStateMachine()) {
+                $this->stateMachine->apply($configuration, $resource);
+            }
```

Comment:
`$configuration->hasStateMachine()` away return `true` when we try to update form even though it has no transition.